uefi/src/proto/hii/config_str.rs

```diff
@@ -169,11 +170,15 @@ impl ConfigurationString {
         if data.len() % 2 != 0 {
             return None;
         }
-        let mut data: Vec<_> = Self::parse_bytes_from_hex(data).collect();
-        data.chunks_exact_mut(2).for_each(|c| c.swap(0, 1));
-        data.extend_from_slice(&[0, 0]);
+        let size_bytes = data.len() / 2 + 2; // includes \0 terminator
```

Comment:
could it be that `data.len()` is for some reason not a multiple of two? In that case `.div_ceil()` would be more fail-safe?

ah nvm, we have a check a few lines above